ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -284,6 +299,40 @@ const NetworksForm = ({
     isEditing,
   ]);
 
+  const newOrderNetworks = () => {
+    if (!orderedNetworksList || orderedNetworksList.length === 0) {
+      return nonTestNetworks;
+    }
+
+    // Create a mapping of chainId to index in orderedNetworksList
+    const orderedIndexMap = {};
+    orderedNetworksList.forEach((network, index) => {
+      orderedIndexMap[`${network.networkId}_${network.networkRpcUrl}`] = index;
+    });
+
+    // Sort nonTestNetworks based on the order in orderedNetworksList
+    const sortedNonTestNetworks = nonTestNetworks.sort((a, b) => {
+      const keyA = `${a.chainId}_${a.rpcUrl}`;
+      const keyB = `${b.chainId}_${b.rpcUrl}`;
+      return orderedIndexMap[keyA] - orderedIndexMap[keyB];
+    });
+
+    return sortedNonTestNetworks;
+  };
+
```

Comment:
nit: related to my other comment suggest suggesting the 'on' and 'handle' convention, espacially for click handlers, maybe handleEditNetworkClick